src/main/java/com/minecolonies/coremod/tileentities/TileEntityColonyBuilding.java

```diff
@@ -409,35 +409,40 @@ public boolean hasCapability(@NotNull final Capability<?> capability, final Enum
+                final World world = colony.getWorld();
+                if (world != null)
+                {
+                    //Add additional containers
+                    providers.addAll(building.getAdditionalCountainers().stream()
+                                       .map(world::getTileEntity)
+                                       .collect(Collectors.toSet()));
+                    providers.removeIf(Objects::isNull);
+                }
 
-            final List<IItemHandler> handlers  = providers.stream()
-                             .flatMap(provider -> InventoryUtils.getItemHandlersFromProvider(provider).stream())
-                             .collect(Collectors.toList());
-            final T cap = super.getCapability(capability, facing);
-            if (cap instanceof IItemHandler)
-            {
-                handlers.add((IItemHandler) cap);
-            }
+                final List<IItemHandler> handlers = providers.stream()
+                                                      .flatMap(provider -> InventoryUtils.getItemHandlersFromProvider(provider).stream())
+                                                      .collect(Collectors.toList());
+                final T cap = super.getCapability(capability, facing);
+                if (cap instanceof IItemHandler)
+                {
+                    handlers.add((IItemHandler) cap);
+                }
 
-            return (T) new CombinedItemHandler(building.getSchematicName(), handlers.stream()
-                                                                     .map(handler -> (IItemHandlerModifiable) handler)
-                                                                              .distinct()
-                                                                              .filter(handler -> handler instanceof IItemHandlerModifiable && handler.getSlots() >= MIN_SLOTS_FOR_RECOGNITION)
-                                                                              .toArray(IItemHandlerModifiable[]::new));
+                return (T) new CombinedItemHandler(building.getSchematicName(), handlers.stream()
+                                                                                  .map(handler -> (IItemHandlerModifiable) handler)
+                                                                                  .distinct()
+                                                                                  .filter(handler -> handler instanceof IItemHandlerModifiable
+                                                                                                       && handler.getSlots() >= MIN_SLOTS_FOR_RECOGNITION)
+                                                                                  .toArray(IItemHandlerModifiable[]::new));
+            }
+            return super.getCapability(capability, facing);
         }
-        return super.getCapability(capability, facing);
+        return null;
```

Comment:
Never return null on the capability. Perform the building != null check in the has capability method and remove the check here.